packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,467 @@
+        this.artifactInfo.modelDir,
+        'model',
+      );
+    } catch (err) {
+      return this.exit(err);
+    }
+
+    if (this.options.model) {
+      debug(`Model name received from command line: ${this.options.model}`);
+
+      this.options.model = utils.toClassName(this.options.model);
+      // assign the model name from the command line only if it is valid
+      if (
+        modelList &&
+        modelList.length > 0 &&
+        modelList.includes(this.options.model)
+      ) {
+        Object.assign(this.artifactInfo, {modelNameList: [this.options.model]});
+      } else {
+        modelList = [];
+      }
+    }
+
+    if (modelList.length === 0) {
+      return this.exit(
+        new Error(
+          `${ERROR_NO_MODELS_FOUND} ${this.artifactInfo.modelDir}.
+        ${chalk.yellow(
+          'Please visit https://loopback.io/doc/en/lb4/Model-generator.html for information on how models are discovered',
+        )}`,
+        ),
+      );
+    }
+
+    return this.prompt([
+      {
+        type: 'checkbox',
+        name: 'modelNameList',
+        message: PROMPT_MESSAGE_MODEL,
+        choices: modelList,
+        when: this.artifactInfo.modelNameList === undefined,
+      },
+    ])
+      .then(props => {
+        Object.assign(this.artifactInfo, props);
+        debug(`props after model list prompt: ${inspect(props)}`);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during model list prompt: ${err}`);
+        return this.exit(err);
+      });
+  }
+
+  async promptModelId() {
+    if (this.shouldExit()) return false;
+    let idType;
+
+    debug(`Model ID property name from command line: ${this.options.id}`);
+    debug(`Selected Models: ${this.artifactInfo.modelNameList}`);
+
+    if (_.isEmpty(this.artifactInfo.modelNameList)) {
+      return this.exit(new Error(`${ERROR_NO_MODEL_SELECTED}`));
+    } else {
+      // iterate thru each selected model, infer or ask for the ID type
+      for (let item of this.artifactInfo.modelNameList) {
+        this.artifactInfo.modelName = item;
+
+        const prompts = [
+          {
+            type: 'input',
+            name: 'propertyName',
+            message: `Please enter the name of the ID property for ${item}:`,
+            default: 'id',
+          },
+        ];
+
+        // user supplied the id from the command line
+        if (this.options.id) {
+          debug(`passing thru this.options.id with value : ${this.options.id}`);
+
+          idType = this.options.id;
+          /**  make sure it is only used once, in case user selected more
+           * than one model.
+           */
+          delete this.options.id;
+        } else {
+          idType = await this._getModelIdType(item);
+          if (idType === null) {
+            const answer = await this.prompt(prompts);
+            idType = answer.propertyName;
+          }
+        }
+        this.artifactInfo.idType = idType;
```

Comment:
nitpick: Can we rename this to `idProp` or `idProperty` -- While looking at the template I was very confused because `idType` to me implied this was a type .. such as `string`, `number`, etc. but IIUC it is the name of the id property. --- I'd say it only needs to be changed on `artifactInfo` and in the `*.ejs` templates.

Reply:
I changed it to idProperty and also change the method named to getModelIdProperty.